_targets.R

```diff
@@ -1,42 +1,129 @@
 library(targets)
 source("1_fetch/src/get_nwis_data.R")
+source("2_process/src/bind_data.R")
 source("2_process/src/process_and_style.R")
 source("3_visualize/src/plot_timeseries.R")
 
 options(tidyverse.quiet = TRUE)
-tar_option_set(packages = c("tidyverse", "dataRetrieval")) # Loading tidyverse because we need dplyr, ggplot2, readr, stringr, and purrr
 
+tar_option_set(packages = c(
+  "tidyverse", 
+  "dataRetrieval",
+  "utils",
+  "futile.logger"
+  )
+)
+
+# site number list 
+site_list <- c("01427207", "01432160", "01436690", "01466500")
+
+# establish common start date, end date, and parameter code for queries
+startDate <- "2014-05-01" 
+endDate <- "2015-05-01"
+parameterCd <- '00010'
+
+# Create output directories if not included
+if(!dir.exists('1_fetch/out/'))
+  dir.create(path = '1_fetch/out/')
+if(!dir.exists('2_process/out/'))
+  dir.create(path = '2_process/out/')
+if(!dir.exists('3_visualize/out/'))
+  dir.create(path = '3_visualize/out/')
+
+
+
+# Download site data and compile files
 p1_targets_list <- list(
   tar_target(
-    site_data,
-    download_nwis_data(),
+    site_1_data,
+    download_nwis_data(
+      site_list[1],
+      "1_fetch/out/", # out directory
+      startDate,
+      endDate,
+      parameterCd
+    ),
+  format = "file"
+  ),
+  tar_target(
+    site_2_data,
+    download_nwis_data(
+      site_list[2],
+      "1_fetch/out/", # out directory
+      startDate,
+      endDate,
+      parameterCd
+    ),
+    format = "file"
   ),
   tar_target(
-    site_info_csv,
-    nwis_site_info(fileout = "1_fetch/out/site_info.csv", site_data),
+    site_3_data,
+    download_nwis_data(
+      site_list[3],
+      "1_fetch/out/", # out directory
+      startDate,
+      endDate,
+      parameterCd
+    ),
+    format = "file"
+  ),
+  tar_target(
+    site_4_data,
+    download_nwis_data(
+      site_list[4],
+      "1_fetch/out/", # out directory
+      startDate,
+      endDate,
+      parameterCd
+    ),
+    format = "file"
+  ),
+  tar_target(
+    compiled_nwis_data,
+    bind_data(
+      site_1_data,
+      site_2_data,
+      site_3_data,
+      site_4_data,
+      "nwisdata", # specify which files to combine, filename search string
+      "2_process/out/all_nwisdata.csv" # file outpath
+    ),
+  format = "file"
+  ),
+  tar_target(
+    compiled_site_info,
```

Comment:
If we adjust with a previous comment, then you can turn this into something like

```r
tar_target(
    compiled_site_info,
    download_nwis_site_info(
        "2_process/out/all_siteinfo.csv", # file outpath
        compiled_nwis_data # file in path
    ),
    format = "file"
)
```